src/saturn_engine/worker/topics/logger.py

```diff
@@ -0,0 +1,56 @@
+from typing import Any
+from typing import AsyncGenerator
+from typing import Callable
+from typing import Optional
+
+import dataclasses
+import json
+from enum import Enum
+
+from saturn_engine.core import TopicMessage
+from saturn_engine.utils.log import getLogger
+from saturn_engine.worker.topic import Topic
+
+
+class LoggingTopic(Topic):
+    """A topic contains default loggers for error handling
+    such as ignoring the error and logging it in debug.
+    """
+
+    class Channels(Enum):
+        IGNORE = "ignore"
+        DEBUG = "debug"
+        INFO = "info"
+        WARNING = "warning"
+        ERROR = "error"
+
+    @dataclasses.dataclass
+    class Options:
+        channel: Optional[str] = None
+
+    def resolve_channel_to_fn(self) -> Callable[[Any], None]:
```

Comment:
you can use `logger.log(level, msg)`, then you just need to map your enum to logging level. You could actually do that straight in the enum:

```
class Level(Enum):
  IGNORE = None
  DEBUG = logging.DEBUG
  ...
```

```
if self.options.level is None:
  return
self.logger.log(self.options.level, ...)
```

Reply:
Ok so I've changed this a bit. The `level` attibute inside `LoggingTopic.Options` is a string in order to be abke to parse the level from our yaml definitions. It then uses a mapping and stores the actual level inside of `LoggingTopic`

Reply:
I'm pretty sure you can serialize enum from the options:

```python
class Level(enum.Enum):
  INFO = "INFO"
class Options:
   level: Level
```

```yaml
options:
  level: INFO
```

Did you try?

Reply:
Ok so now it uses the enum entirely. When passing the value to `logging.Logger.log` I use the function `logging.getLevelName(self.level.value)` to resolve it to an int, which is required by `logging.Logger.log`.